### DISCUSSION TITLE
coqui-ai trainer vs Lightning

### DISCUSSION THREAD

**Discussion by amitaie** [coqui-ai/Trainer#80] [Q&A]
Hey, the idea of 'trainers' is new to me and i'm struggling to understand the different between  coqui-ai trainer and Lightning. what the benefits of using coqui-ai trainer?

thank you,

**Comment by erogol** [comment]
I've not checked the lighting for a while but 

- Simplicity: Just a single file for better debugging, and editing. 
- Logging: You can easily tie to one of the well-known loggers (Tensorboard, ClearML or your own)
- Reproducibility: https://erogol.com/2022/10/01/todos-for-effective-ml-team-work-at-an-early-stage-startup#keep-things-reproducible
- Config-based management: We use Coqpit as the core way to interact with the model and the trainer. 

In short, if you train large models without worrying about the trainer at all go with lightning but if you have the concerns above, then 👟 is for you.
